examples/jsm/helpers/OctreeHelper.js

```diff
@@ -39,17 +53,9 @@ class OctreeHelper extends LineSegments {
 
 		}
 
-		traverse( octree.subTrees );
-
-		const geometry = new BufferGeometry();
-		geometry.setAttribute( 'position', new Float32BufferAttribute( vertices, 3 ) );
-
-		super( geometry, new LineBasicMaterial( { color: color, toneMapped: false } ) );
+		traverse( this.octree.subTrees );
 
-		this.octree = octree;
-		this.color = color;
-
-		this.type = 'OctreeHelper';
+		this.geometry.setAttribute( 'position', new Float32BufferAttribute( vertices, 3 ) );
```

Comment:
There is one issue with this implementation. Since there is now an explicit `update()` method, you potentially overwrite the `position` buffer attribute over and over again. This is not right. Instead of overwriting the attribute, it's more correct to just update the existing values.

Depending on the Octree, the resulting attribute might change in size so updating is not always possible. So you have to call `dispose()` on the geometry and create a new one with the new position attribute.

Reply:
No, you can only check the number of geometries used but not the individual buffers. 

Update: Since `WebGLAttributes` uses a weak map, there should not be a memory leak. However, it is still better to not just overwrite the buffer and hope for the GC but explicitly free resources via `dispose()`.